### PR TITLE
PullRequest.setMilestone should use the Issue API

### LINKED ISSUE #416: Can only set PR milestone through instance of GHIssue, not GHPullRequest
[`GHPullRequest.setMilestone()`](https://github.com/kohsuke/github-api/blob/master/src/main/java/org/kohsuke/github/GHIssue.java#L206) which was helpfully added in https://github.com/kohsuke/github-api/pull/397 appears to have a minor bug.  It works when called on an instance of `GHIssue` directly but not on `GHPullRequest`.  I **think** this is because it calls `edit` rather than `editIssue` which is used by labels, assignees, etc and so it gets the wrong API endpoint for setting a milestone which is a property of the issue not the pull request.

I would submit a fix for this myself, except: https://github.com/kohsuke/github-api/issues/415.  Sorry!

### PR DESCRIPTION
**Pull request by gastaldi**
# Description 
When `PullRequest.setMilestone` is called, the Issue API must be used to update the assigned milestone.
Fixes #416
Fixes https://github.com/quarkusio/quarkus-backports/issues/2 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [X] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [X] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [X] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [X] Fill in the "Description" above. 
- [X] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by gastaldi** [comment]
I followed the steps in `CONTRIBUTING.md` to generate the wiremock data but it still doesn't work. I am getting 

```
[ERROR] testUnsetMilestone(org.kohsuke.github.GHPullRequestTest)  Time elapsed: 3.172 s  <<< ERROR!
org.kohsuke.github.GHFileNotFoundException: http://localhost:46503/repos/hub4j-test-org/github-api/milestones {"message":"Not Found","documentation_url":"https://developer.github.com/v3/issues/milestones/#create-a-milestone"}
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:493)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:421)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:364)
	at org.kohsuke.github.Requester.fetch(Requester.java:74)
	at org.kohsuke.github.GHRepository.createMilestone(GHRepository.java:2428)
	at org.kohsuke.github.GHPullRequestTest.testUnsetMilestone(GHPullRequestTest.java:330)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.kohsuke.github.junit.WireMockMultiServerRule$1.evaluate(WireMockMultiServerRule.java:69)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.io.FileNotFoundException: http://localhost:46503/repos/hub4j-test-org/github-api/milestones
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1969)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1964)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1963)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1531)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient$HttpURLConnectionResponseInfo.bodyStream(GitHubHttpUrlConnectionClient.java:197)
	at org.kohsuke.github.GitHubResponse$ResponseInfo.getBodyAsString(GitHubResponse.java:321)
	at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:86)
	at org.kohsuke.github.Requester.lambda$fetch$1(Requester.java:74)
	at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:462)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:413)
	... 35 more
Caused by: java.io.FileNotFoundException: http://localhost:46503/repos/hub4j-test-org/github-api/milestones
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1915)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient.getResponseInfo(GitHubHttpUrlConnectionClient.java:69)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:401)
	... 35 more
```

What am I doing wrong?

**Comment by gastaldi** [comment]
Can this fix be merged and released soon @bitwiseman?

**Comment by bitwiseman** [comment]
@gastaldi 
I've invited you to the `hub4j-test-org`.  Could you redo your test and recording using that org and using `getTempRepository()`.  That way we can re-record the the tests later if needed.

File: src/main/java/org/kohsuke/github/GitHub.java [hunk 1]
```diff
@@ -543,6 +543,9 @@ public GHOrganization getOrganization(String name) throws IOException {
      */
     public GHRepository getRepository(String name) throws IOException {
         String[] tokens = name.split("/");
+        if (tokens.length < 2) {
+            throw new IllegalArgumentException("Repository name must be in format owner/reponame");
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitHub.java:547]:
```suggestion
            throw new IllegalArgumentException("Repository name must be in format owner/repo");
```

GitHub docs all use `repo` for this.

**Comment by gastaldi** [comment]
Done, thanks!

**Comment by bitwiseman** [comment]
@gastaldi 
We tend to do releases every couple of weeks or once we have a chunk of PRs merged. I'm happy to publish a new release in a few days. Okay?

**Comment by gastaldi** [comment]
Fine to me. Thanks!

**Pull request merged**